packages/core/src/AbstractAppliance.js

```diff
@@ -38,40 +37,69 @@ class AbstractAppliance extends IAppliance {
 		)
 	}
 
-	/**
-	 * Called by a third party to pass data into the appliance.
-	 * If the payload is valid, it is added to the buffer and the appliance is invoked.
-	 *
-	 * @throws
-	 * @param  {[type]} payload [description]
-	 * @return {[type]}         [description]
-	 */
 	ingestPayload = async (payload) => {
-		this.emitter.emit(INPUT_RECEIVED, { payload })
-		assert(isPayloadInstance(payload), 'You cannot ingest data that is not an instance of Payload.')
-		assert(await this.isValidPayload(payload), 'Payload does not satisfy appliance ingestion conditions.')
-		this.payloadBuffer.add(payload)
-		return this.invoke()
+		assert(
+			Payload.isPayload(payload),
+			'You cannot ingest data that is not an instance of Payload.',
```

Comment:
Okay, so I think the thinking here is error events are for runtime errors that can be handled, the asserts are for catching errors in construction. Correct?

Reply:
That's a good distinction -- given that, this should actually be an emitted error rather than an assertion I believe... right?

Reply:
Not sure. I stumbled on [this post](https://www.joyent.com/node-js/production/design/errors#operational-errors-vs-programmer-errors) about operational vs. programmer errors yesterday. I think these are programmer errors? And therefore it's okay to throw an assertion error?

Reply:
Keeping it as is for now; that logic is sound (🤔 probably worth documenting somewhere)